### DISCUSSION TITLE
CSV import Orgchart - problem with connector points

### DISCUSSION THREAD

**Discussion by tojeffraymond** [jgraph/drawio#3208] [Q&A]
Hi,

Working with draw.io desktop.  I presume my question is also relevant to draw.io webapp.
Looked in desktop repo to see if my topic was discussed.  Which is not.
----

In CSV import, with layout set to "auto", drawio uses the generic layout (which I think is the mxHierarchicalLayout).
I've put entry/exit point for the connect object (relation) since my objective is not to have mxpoint(waypoints). 
I'll explain why later...

From the input of the CSV...
`# connect: {"from":"refs", "to":"id", "invert":true, "style":"curved=0;endArrow=blockThin;endFill=1;exitDx=0;exitDy=0;entryDx=0;entryDy=0;entryX=0;entryY=0.5;edgeStyle=orthogonalEdgeStyle;"}`

After the import, the setting for entry/exit points of the connectors(relations) is kept. Which is expected and ok.

But when I change my layout to orgchart(SingleRightColumn), my setting is changed and mxpoints are added to the connectors.

An example :
Before the change of layout
```
<mxCell id="0NidjX7elacmojV8UP3N-9" value="" style="curved=0;endArrow=blockThin;endFill=1;exitDx=0;exitDy=0;entryDx=0;entryDy=0;entryX=0;entryY=0.5;edgeStyle=orthogonalEdgeStyle;" edge="1" parent="1" source="0NidjX7elacmojV8UP3N-7" target="0NidjX7elacmojV8UP3N-8">
      <mxGeometry relative="1" as="geometry" />
    </mxCell>
```
After the change to Orgchart layout
```
<mxCell id="0NidjX7elacmojV8UP3N-9" value="" style="curved=0;endArrow=blockThin;endFill=1;exitDx=0;exitDy=0;entryDx=0;entryDy=0;edgeStyle=orthogonalEdgeStyle;" edge="1" parent="1" source="0NidjX7elacmojV8UP3N-7" target="0NidjX7elacmojV8UP3N-8">
      <mxGeometry relative="1" as="geometry">
        <Array as="points">
          <mxPoint x="212.5" y="45" />
          <mxPoint x="212.5" y="77.5" />
          <mxPoint x="237.5" y="77.5" />
        </Array>
      </mxGeometry>
    </mxCell>
```

The problem here is that, after the import and the layout set to orgchart (SingleRightColum), I need to realign my shapes. 

![image](https://user-images.githubusercontent.com/46726829/204072737-caea5685-f9c3-43d2-a8b4-16f61a6e91a4.png)

So, when I align them, make changes to the text or move the shapes, since the connectors are not attached to the connection points of the shapes, it becomes very painful to work with the diagram. It's time consuming...

![image](https://user-images.githubusercontent.com/46726829/204072858-f5339653-bc37-48a9-9532-92df3d9e3109.png)

I'm wondering why the app is not using the native connection points of the shapes.

Any thoughts ?
Should it be an improvement request ? I guess... ;-)

**Comment by trovaleRyron** [comment]
Try using floating connectors instead of fixed. 

https://www.youtube.com/watch?v=r6QzW61ZSFc&ab_channel=draw.io
